lib/spack/spack/repository.py

```diff
@@ -994,8 +994,8 @@ def _get_pkg_module(self, pkg_name):
                 # SyntaxError strips the path from the filename so we need to
                 # manually construct the error message in order to give the
                 # user the correct package.py where the syntax error is located
-                raise SyntaxError('invalid syntax in {0:}, line {1:}'
-                                  ''.format(file_path, e.lineno))
+                raise SyntaxError('{0:} in {1:}, line {2:}'
+                                  ''.format(str(e), file_path, e.lineno))
```

Comment:
Those colons and the string conversion don't actually do anything. I'm also not sure how I feel about that string line break. It might look better as:
```python
msg = '{0} in {1}, line {2}'.format(e, file_path, e.lineno))    
raise SyntaxError(msg)
```

Reply:
I added the error construction in #6458, before the path was lacking from `package.py`. Don't ask me why I put the colons there, probably because I added the python2.6 compatibility later on..